src/credentials/providers/credentialsProviderManager.ts

```diff
@@ -20,9 +19,13 @@ export class CredentialsProviderManager {
 
         for (const factory of this.providerFactories) {
             await factory.refresh()
-
             const refreshed = factory.listProviders()
-            recordAwsLoadCredentials({ credentialSourceId: factory.getCredentialType(), value: refreshed.length })
+            const providerType = factory.getProviderType()
+            if (!providerType) {
+                continue
+            }
+            const telemType = credentialsProviderToTelemetryType(providerType)
+            recordAwsLoadCredentials({ credentialSourceId: telemType, value: refreshed.length })
```

Comment:
Introduced `credentialsProviderToTelemetryType()` because the "credential source type" that we define is not one-to-one with the telemetry kinds. See https://github.com/aws/aws-toolkit-vscode/pull/1725#discussion_r626204148